code/__HELPERS/lists.dm

```diff
@@ -424,5 +424,7 @@
 #define LAZYINITLIST(L) if (!L) L = list()
 
 #define UNSETEMPTY(L) if (L && !L.len) L = null
-
+#define LAZYREMOVE(L, I) if(L) { L -= I; if(!L.len) { L = null; } }
+#define LAZYADD(L, I) if(!L) { L = list(); } L += I;
+#define LAZYACCESS(L, I) (L ? (isnum(I) ? (I > 0 && I <= L.len ? L[I] : null) : L[I]) : null)
 #define LAZYLEN(L) ( L ? L.len : 0 )

```

Comment:
MSO did some testing
http://pastebin.com/tA0MGUr3

apparently length is faster